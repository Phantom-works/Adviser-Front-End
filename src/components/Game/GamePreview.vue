<template>
  <div class="game-preview">
    <img :alt="game.title" :src="game.imageURL" class="game-image"/>
  </div>
</template>

<script lang="ts" setup>
import Game from '@/Model/Game'

defineProps<{ game: Game }>()
</script>

<style scoped>
.game-preview {
  display: flex;
  flex-wrap: wrap;
  align-items: self-start;
  justify-content: center;
  width: 200px;
  height: 300px;
  border-radius: 10px;
  margin: 0 1rem 1rem;
}

.game-image {
  object-fit: cover; /* add object-fit */
  width: 100%; /* adjust width */
  height: 100%; /* adjust height */
  border-radius: 10px; /* add border-radius */
}
</style>

