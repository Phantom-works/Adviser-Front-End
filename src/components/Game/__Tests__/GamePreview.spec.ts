import { describe, expect, it } from 'vitest'
import { mount } from '@vue/test-utils'
import GamePreview from '../GamePreview.vue'

describe('GamePreview Snapshots', () => {
  // Arrange

  it('Snapshot Empty Game', () => {
    //Arrange
    const game = {}
    // Acts
    const wrapper = mount(GamePreview, {
      props: {
        game: game
      }
    })
    // Assert
    expect(wrapper.html()).toMatchSnapshot()
  })
})
