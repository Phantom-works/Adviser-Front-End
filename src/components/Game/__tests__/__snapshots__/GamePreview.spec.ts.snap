// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`GamePreview Snapshots > Snapshot Empty Game 1`] = `"<div data-v-023eec1f=\\"\\" class=\\"game-preview\\"><img data-v-023eec1f=\\"\\" class=\\"game-image\\"></div>"`;

exports[`GamePreview Snapshots > Snapshot Game empty 1`] = `"<div data-v-023eec1f=\\"\\" class=\\"game-preview\\"><img data-v-023eec1f=\\"\\" alt=\\"\\" src=\\"\\" class=\\"game-image\\"></div>"`;

exports[`GamePreview Snapshots > Snapshot Game factorio 1`] = `"<div data-v-023eec1f=\\"\\" class=\\"game-preview\\"><img data-v-023eec1f=\\"\\" alt=\\"factorio\\" src=\\"cool factorio image\\" class=\\"game-image\\"></div>"`;

exports[`GamePreview Snapshots > Snapshot Game long title 1`] = `"<div data-v-023eec1f=\\"\\" class=\\"game-preview\\"><img data-v-023eec1f=\\"\\" alt=\\"3A45ST6JYDK6U4TAW4RES5T6JDYK7E4YRAUS5TI6DR4YEU5SRTID6Y\\" src=\\"\\" class=\\"game-image\\"></div>"`;

exports[`GamePreview Snapshots > Snapshot no game 1`] = `"<div data-v-023eec1f=\\"\\" class=\\"game-preview\\"><img data-v-023eec1f=\\"\\" alt=\\"factorio\\" src=\\"cool factorio image\\" class=\\"game-image\\"></div>"`;
